src/utils/vectorEncoder.ts

```diff
@@ -0,0 +1,398 @@
+ * The algorithms used to encode the data are describe in:
+ *            http://www.iupac.org/publications/pac/pdf/2001/pdf/7311x1765.pdf
+ */
+const newLine = '\r\n';
+
+const pseudoDigits: string[][] = [
+  ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'],
+  ['@', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
+  ['@', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i'],
+  ['%', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R'],
+  ['%', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r'],
+  [' ', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 's'],
+];
+
+const SQZ_P = 1;
+const SQZ_N = 2;
+const DIF_P = 3;
+const DIF_N = 4;
+const DUP = 5;
+const MaxLinelength = 100;
+
+/**
+ * This function encodes the given vector. The encoding format is specified by the
+ * encoding option
+ * @param {Array} data
+ * @param {number} firstX
+ * @param {number} intervalX
+ * @param {string} encoding: ('FIX','SQZ','DIF','DIFDUP','CVS','PAC') Default 'DIFDUP'
+ * @return {string}
+ */
+export function encode(
+  data: DoubleArray,
+  firstX: number,
+  intervalX: number,
+  encoding: string,
+) {
+  switch (encoding) {
+    case 'FIX':
+      return fixEncoding(data, firstX, intervalX);
+    case 'SQZ':
+      return squeezedEncoding(data, firstX, intervalX);
+    case 'DIF':
+      return differenceEncoding(data, firstX, intervalX);
+    case 'DIFDUP':
+      return differenceDuplicateEncoding(data, firstX, intervalX);
+    case 'CSV':
+      return commaSeparatedValuesEncoding(data, firstX, intervalX);
+    case 'PAC':
+      return packedEncoding(data, firstX, intervalX);
+    default:
+      return differenceEncoding(data, firstX, intervalX);
+  }
+}
+
+/**
+ * @private
+ * No data compression used. The data is separated by a comma(',').
+ * @param data
+ * @param firstX
+ * @param intervalX
+ * @return {string}
+ */
+export function commaSeparatedValuesEncoding(
+  data: DoubleArray,
+  firstX: number,
+  intervalX: number,
+) {
+  return fixEncoding(data, firstX, intervalX, ',');
+}
+
+/**
+ * @private
+ * No data compression used. The data is separated by the specified separator.
+ * @param  data
+ * @param firstX
+ * @param intervalX
+ * @param  separator, The separator character
+ * @return {string}
+ */
+export function fixEncoding(
+  data: DoubleArray,
+  firstX: number,
+  intervalX: number,
+  separator = ' ',
+) {
+  let outputData = '';
+  let j = 0;
+  let TD = data.length;
+  let i;
+  while (j < TD - 7) {
+    outputData += Math.ceil(firstX + j * intervalX);
+    for (i = 0; i < 8; i++) {
+      outputData += separator + data[j++];
+    }
+    outputData += newLine;
+  }
+  if (j < TD) {
+    // We add last numbers
+    outputData += Math.ceil(firstX + j * intervalX);
+    for (i = j; i < TD; i++) {
+      outputData += separator + data[i];
+    }
+  }
+  return outputData;
+}
+
+/**
+ * @private
+ * No data compression used. The data is separated by the sign of the number.
+ * @param {Array} data
```

Comment:
You should not specify the type in TS. For the documentation the type is extracted from the function parameters